src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,148 @@
+package duke;
+
+import java.util.zip.DataFormatException;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ListCommand;
+
+
+/**
+ * Understands inputs given by user and validates them.
+ */
+public class Parser {
+
+    /**
+     * Understands input from user
+     *
+     * @param userText input from user
+     * @param taskList list of tasks user has currently
+     * @return boolean value of whether conversation is to be continued
+     */
+    public static boolean understandText(String userText, TaskList taskList) {
```

Comment:
As the method returns a boolean, more intuitive name, such as canUnderstand, can be used.